src/main/scala/org/broadinstitute/dsde/workbench/sam/api/SecurityDirectives.scala

```diff
@@ -10,23 +10,24 @@ import org.broadinstitute.dsde.workbench.sam.service.PolicyEvaluatorService
 import ImplicitConversions.ioOnSuccessMagnet
 import cats.implicits._
 import cats.effect.IO
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
 
 trait SecurityDirectives {
   def policyEvaluatorService: PolicyEvaluatorService
 
-  def requireAction(resource: FullyQualifiedResourceId, action: ResourceAction, userId: WorkbenchUserId): Directive0 =
+  def requireAction(resource: FullyQualifiedResourceId, action: ResourceAction, userId: WorkbenchUserId, samRequestContext: SamRequestContext = SamRequestContext(null)): Directive0 =
     requireOneOfAction(resource, Set(action), userId)
 
-  def requireOneOfAction(resource: FullyQualifiedResourceId, requestedActions: Set[ResourceAction], userId: WorkbenchUserId): Directive0 =
+  def requireOneOfAction(resource: FullyQualifiedResourceId, requestedActions: Set[ResourceAction], userId: WorkbenchUserId, samRequestContext: SamRequestContext = SamRequestContext(null)): Directive0 =
     Directives.mapInnerRoute { innerRoute =>
-      onSuccess(hasPermissionOneOf(resource, requestedActions, userId)) { hasPermission =>
+      onSuccess(hasPermissionOneOf(resource, requestedActions, userId, samRequestContext)) { hasPermission =>
```

Comment:
Do we want this to be a part of the trace? Trying to add tracing by changing the order leads to some errors (which may be easily solved if this is the behavior we want)

when reordering the code:

from
```
    requireAction(managedGroup, SamResourceActions.readPolicy(accessPolicyName), userInfo.userId) {
      completeWithTrace(samRequestContext =>
        managedGroupService.listPolicyMemberEmails(managedGroup.resourceId, accessPolicyName, samRequestContext).map(x => StatusCodes.OK -> x.toSet))
    }
```
to
```
    completeWithTrace { samRequestContext =>
      requireAction(managedGroup, SamResourceActions.readPolicy(accessPolicyName), userInfo.userId) {
        managedGroupService.listPolicyMemberEmails(managedGroup.resourceId, accessPolicyName, samRequestContext).map(x => StatusCodes.OK -> x.toSet)
      }
    }
```
results in:
```
[error] /Users/ajang/Firecloud/sam/src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ManagedGroupRoutes.scala:104:117: type mismatch;
[error]  found   : cats.effect.IO[(akka.http.scaladsl.model.StatusCodes.Success, scala.collection.immutable.Set[org.broadinstitute.dsde.workbench.model.WorkbenchEmail])]
[error]  required: akka.http.scaladsl.server.Route
[error]     (which expands to)  akka.http.scaladsl.server.RequestContext => scala.concurrent.Future[akka.http.scaladsl.server.RouteResult]
[error]         managedGroupService.listPolicyMemberEmails(managedGroup.resourceId, accessPolicyName, samRequestContext).map(x => StatusCodes.OK -> x.toSet)
[error]                                                                                                                     ^
[error] one error found
```

Reply:
//implemented